packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -439,189 +276,30 @@ describe('NetworkController', () => {
                 },
               },
             ]);
-            createMetamaskProviderMock.mockReturnValue(fakeMetamaskProvider);
+            const fakeNetworkClient = buildFakeNetworkClient(fakeProvider);
+            createNetworkClientMock.mockReturnValue(fakeNetworkClient);
 
             await controller.initializeProvider();
 
-            expect(createMetamaskProviderMock).toHaveBeenCalledWith({
-              chainId: undefined,
-              engineParams: { pollingInterval: 12000 },
-              nickname: undefined,
+            expect(createNetworkClientMock).toHaveBeenCalledWith({
               rpcUrl: 'http://localhost:8545',
-              ticker: undefined,
+              chainId: toHex(1337),
+              type: NetworkClientType.Custom,
             });
             const { provider } = controller.getProviderAndBlockTracker();
+            assert(provider, 'Block tracker is somehow unset');
             const promisifiedSendAsync = promisify(provider.sendAsync).bind(
               provider,
             );
             const chainIdResult = await promisifiedSendAsync({
+              id: '1',
```

Comment:
Because we have a proper type for the provider, that means we have a proper type for `promisifiedSendAsync`. It expects a fully qualified JSON-RPC request object, so we must update all instances where we are calling `promisifiedSendAsync` to match. (This is the case on the extension side as well.)